app/code/Magento/Eav/Model/Config.php

```diff
@@ -583,7 +593,7 @@ private function initSystemAttributes($entityType, $systemAttributes)
             return;
         }
 
-        $cacheKey = self::ATTRIBUTES_CACHE_ID . '-' . $entityTypeCode . '-preload';
+        $cacheKey = self::ATTRIBUTES_CACHE_ID . '-' . $this->getStoreId() . '-' . $entityTypeCode . '-preload';
```

Comment:
Can the system attributes be separated by the store? If not - I think we can use a "global" value

Reply:
Method has the name initSystemAttributes but not only system attributes being loaded by it.
It loads attribute provided in attributesForPreload argument which populated from di configuration.
So if we not split here we will get issue described in the pull request again.

But good news that we can remove splitting from cacheUserDefinedAttribute method.

Reply:
I suggest to rename private initSystemAttributes method to initPreloadAttributes.